lib/utils/dns.go

```diff
@@ -89,7 +89,8 @@ func (d *DNSConfig) String() string {
 // See resolv.conf(5) on a Linux machine.
 // TODO(rsc): Supposed to call uname() and chop the beginning
 // of the host name to get the default search domain.
-func DNSReadConfig(rdr io.Reader) (*DNSConfig, error) {
+// noDefaultNameservers indicates we should not emulate libc behaviour and include a localhost resolver.
+func DNSReadConfig(rdr io.Reader, noDefaultNameservers bool) (*DNSConfig, error) {
```

Comment:
I'm a bit confused about the boolean parameter as well - do we need this or can we fall back to the suggested behavior by default - e.g. not include the defaults at all?

Reply:
This is the approach I attempted to take last year and the review process stalled there as well, so this is my attempt to explicitly indicate that the function violates normal behaviour so that it get's accepted.

https://github.com/gravitational/planet/pull/479

Reply:
The boolean is only used to pass true in the only use-case - why not just remove it and not include the default nameserver or am I missing anything? If this is a way to go I'd be fine with approving https://github.com/gravitational/planet/pull/479 instead.